crates/resolver/src/name_server/name_server.rs

```diff
@@ -94,95 +95,90 @@ impl<C: DnsHandle, P: ConnectionProvider<ConnHandle = C>> NameServer<C, P> {
             // TODO: we need the local EDNS options
             self.state = Arc::new(NameServerState::init(None));
 
+            let (client, bg) = self.conn_provider
+                .new_connection(&self.config, &self.options)
+                .await?;
+
             // establish a new connection
-            self.client = Some(
-                self.conn_provider
-                    .new_connection(&self.config, &self.options),
-            );
+            self.client = Some(client);
+
+            // FIXME:  generify this:
+            if let Some(bg) = bg { spawn(bg.map(|r: Result<(), _>| ())); };
```

Comment:
Ok, this spawn is something that at a minimum needs to be generified, but maybe we can come up with an elegant way to work with these that makes that abstraction higher level.

What if we have a NameServerBackground that is sent over a channel all the ConnectionBackground tasks and is responsible for spawning them? Would that be easier than requiring a generic spawn function to be passed into the AsyncResolver -> NameServerPool -> NameServer (here)?